owlkettle/widgets.nim

```diff
@@ -3401,6 +3401,113 @@ renderable AboutDialog of BaseWidget:
         "Art": @["Max Mustermann"]
       }
 
+type ScalePosition* = enum
+  ScaleLeft,
+  ScaleRight,
+  ScaleTop,
+  ScaleBottom
+
+proc toGtk(pos: ScalePosition): GtkPositionType =
+  result = GtkPositionType(ord(pos))
+
+type ScaleMark* = tuple
+  label: Option[string]
+  value: float64
+  position: ScalePosition
+
+renderable Scale of BaseWidget:
+  ## Wrapper for GTK Scale Widget: https://docs.gtk.org/gtk4/class.Scale.html
+  min: float64 = 0
+  max: float64 = 100
+  value: float64 = 0
+  marks: seq[ScaleMark] = @[]
+  inverted: bool = false
+  showValue: bool = true
+  stepSize: float64 = 5
+  pageSize: float64 = 10
+  orient: Orient = OrientX
+  showFillLevel: bool = true
+  precision: int64 = 1
+  valuePosition: ScalePosition
+  
+  proc valueChanged(newValue: float64) # Emitted when the range value changes.
+
+  hooks:
+    beforeBuild:
+      let orient: Orient = if widget.hasOrient: widget.valOrient else: OrientX
+      state.internalWidget = gtk_scale_new(orient.toGtk(), nil.GtkAdjustment)
+
+    connectEvents:
+      proc valueChangedEventCallback(
+        widget: GtkWidget, 
+        data: ptr EventObj[proc(newValue: float)]
+      ) {.cdecl.} =
+        let scaleValue: float64 = gtk_range_get_value(widget).float64
+        ScaleState(data[].widget).value = scaleValue
+        data[].callback(scaleValue)
+        data[].redraw()
+      
+      state.connect(state.valueChanged, "value-changed", valueChangedEventCallback)
+      
+    disconnectEvents:
+      state.internalWidget.disconnect(state.valueChanged)
+
+  hooks min:
+    property:
+      state.internalWidget.gtk_range_set_range(state.min.cfloat, state.max.cfloat)
+  
+  hooks max:
+    property:
+      state.internalWidget.gtk_range_set_range(state.min.cfloat, state.max.cfloat)
+
+  hooks marks:
+    property:
+      state.internalWidget.gtk_scale_clear_marks()
+      for mark in widget.valMarks:
+        let label: string = if mark.label.isSome(): mark.label.get() else: $mark.value
+        gtk_scale_add_mark(state.internalWidget, mark.value , mark.position.toGtk(), label.cstring)
+
+  hooks value:
+    build:
```

Comment:
I had to add this one back in. If I don't have this build hook, the initial value just doesn't get properly set for value, at least not in my compiles. Can you check if that's also true on your end?

Reply:
I am unable to reproduce this problem.

Though the state should definitely already store the value. This is the generated code for the property hook (generated using `-d:owlkettleDebug`):

```nim
  if widget.hasValue:
    state.value = widget.valValue
  else:
    state.value = 0
  block:
    gtk_range_set_value(state.internalWidget, state.value.cdouble)
```

Reply:
I might have just done something wrong at somewhere, if I just remove the build hook now it works for me as well.
I've thus removed said build hook, we're good.